src/lib/menu/menu-directive.ts

```diff
@@ -81,6 +100,12 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
     }
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['xPosition'] || changes['yPosition']) {
+      this.setPositionClasses();
```

Comment:
Curious why we are adding a hook to call `setPositionClasses` when we already have the setters above. I think it might be cleaner not to break up the logic.